### PR TITLE
manix: 0.6.3 -> 0.7.1

### PR DESCRIPTION
**Pull request by HexSleeves**
## Description of changes

Manix 0.6.3 was unmaintained and far behind on package updates. We submitted issues without any response. This is a fork of manix which just returns it to a usable state for users

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by AndersonTorres** [comment]
So much care for the upstream source code, not so much care for the package.

Restart by reading the [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md) you professedly checked.

**Comment by lecoqjacob** [comment]
> So much care for the upstream source code, not so much care for the package.
> 
> Restart by reading the [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md) you professedly checked.

I read through the doc and followed the instructions. Would you care to elaborate on what I missed? I assume something big by the hostility

**Comment by AndersonTorres** [comment]
> I read through the doc and followed the instructions. Would you care to elaborate on what I missed? I assume something big by the hostility

https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#commit-conventions

- Create a commit for each logical unit.
- Format the commit messages in the following way:

> (pkg-name | nixos/<module>): (from -> to | init at version | refactor | etc)
> (Motivation for change. Link to release notes. Additional information.)

More detailed:

- A commit for adding yourself as a new maintainer, formatted as

```
maintainers: add lecoqjacob
```

- A commit for the change on the package expression itself, formatted as:

```
manix: 0.6.3 -> 0.7.1
```

**Comment by lecoqjacob** [comment]
> > I read through the doc and followed the instructions. Would you care to elaborate on what I missed? I assume something big by the hostility
> 
> 
> 
> https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#commit-conventions
> 
> 
> 
> - Create a commit for each logical unit.
> 
> - Format the commit messages in the following way:
> 
> 
> 
> > (pkg-name | nixos/<module>): (from -> to | init at version | refactor | etc)
> 
> > (Motivation for change. Link to release notes. Additional information.)
> 
> 
> 
> More detailed:
> 
> 
> 
> - A commit for adding yourself as a new maintainer, formatted as
> 
> 
> 
> ```
> 
> maintainers: add lecoqjacob
> 
> ```
> 
> 
> 
> - A commit for the change on the package expression itself, formatted as:
> 
> 
> 
> ```
> 
> manix: 0.6.3 -> 0.7.1
> 
> ```
> 
> 

Thank you. I'll close this issue and restart

**Comment by mrcjkb** [comment]
> Thank you. I'll close this issue and restart

@lecoqjacob no need to close this PR.

You can reset your last 2 commits with

```console
git reset --soft HEAD~2
```

and then stage and commit the chunks individually :smile:

**Comment by lecoqjacob** [comment]
@mrcjkb oh i didnt even know about `-with-lease` i always just used `force`!

**Comment by lecoqjacob** [comment]
@mrcjkb  @AndersonTorres is this ready for review or is it blocked by `23.11` release schedule right now?

File: pkgs/tools/nix/manix/default.nix
Comment (mrcjkb) [inline pkgs/tools/nix/manix/default.nix:16]:
```suggestion
    hash = "sha256-kTQbeOIGG1HmbsXKfXw5yCZ49kGufbGiCkkIRMTwcsg=";
```

**Review by mrcjkb — APPROVED** [review]

**Comment by mrcjkb** [comment]
Result of `nixpkgs-review pr 260061` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>manix</li>
  </ul>
</details>

**Comment by AndersonTorres** [comment]
Squash the two last commits

**Comment by ShaddyDC** [comment]
@lecoqjacob are you still planning on fixing this up? It looks good to me except for the yet-to-be-squashed commits

**Comment by lecoqjacob** [comment]
> @lecoqjacob are you still planning on fixing this up? It looks good to me except for the yet-to-be-squashed commits

Thanks for the reminder. i totally forgot to do this. :/

**Pull request merged**
